tests/php/ORM/HierarchyCachingTest.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace SilverStripe\Versioned\Tests;
+
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\ORM\Hierarchy\Hierarchy;
+use SilverStripe\Versioned\Versioned;
+use SilverStripe\ORM\Tests\HierarchyTest\TestObject;
+use SilverStripe\ORM\Tests\HierarchyTest\HideTestObject;
+use SilverStripe\ORM\Tests\HierarchyTest\HideTestSubObject;
+
+/**
+ * @internal Only test the right values are returned, not that the cache is actually used.
+ */
+class HierachyCacheTest extends SapphireTest
+{
+
+    protected static $fixture_file = 'HierarchyTest.yml';
+
+    protected static $extra_dataobjects = array(
+        TestObject::class,
+        HideTestObject::class,
+        HideTestSubObject::class,
+    );
+
+    public function setUp()
+    {
+        parent::setUp();
+        TestObject::singleton()->flushCache();
+    }
+
+    public static function setUpBeforeClass()
+    {
+        parent::setUpBeforeClass();
+        HideTestObject::config()->update(
```

Comment:
Can you apply this to SiteTree instead?

Reply:
I'd rather note as it ought to work with this and that's how the other Hierarchy test are using.